doc/rst/tools/mason/mason.rst

```diff
@@ -159,202 +194,364 @@ To try out different values at runtime, pass the values for ``number`` to ``maso
-
-To test the project, run ``mason test [options]``, which will update the lock file, compile the tests,
-and produce the binary within ``target/test/`` as follows::
-
-    MyPackage/
-      Mason.toml
-      Mason.lock
-      src/
-        MyPackage.chpl
-      test/
-        sampleTest.chpl
-      target/
-        test/
-          sampleTest
-        debug/
-        (release/)
-          MyPackage
+  MyProject
+   │  
+   ├── Mason.lock
+   ├── Mason.toml
+   ├── example
+   ├── src
+   │   └── myProject.chpl
+   ├── target
+   │   ├── debug
+   │   │   └── myProject
+   │   ├── example
+   │   ├── release
+   │   │   └── myProject
+   │   └── test
+   └── test
+        └── myProjectTest.chpl
 
+Mason testing is based on exit code which means that if the project's tests compile
+and run successfully, despite the "result" of the program, the test's pass. For this
+reason, Mason users should configure their tests such that a failure produces an
+exit code other than 0. Using ``exit()`` is the easiest way to do this, but throwing
+errors is another way to accomplish the same thing.
 
+To run the test(s), use the command ``mason test``. Mason will gather all the tests
+found in ``test/``, compile them with the dependencies listed in your ``Mason.toml``
+and run them producing the following output::
 
-The Manifest File
-=================
+  --- Results ---
+  Test: myProjectTest Passed
+  
+  --- Summary:  1 tests run ---
+  -----> 1 Passed
+  -----> 0 Failed
 
-The ``Mason.toml`` manifest file is written in TOML(for more information see TOML section below).
-Each time a new project is created in Mason a standard TOML file is included in the top-level
-directory of the project.
+If the standard output of the tests is desired, simply throw the ``--show`` flag.
+The output of ``mason test --show`` in this case would be::
 
-For example, ``Mason.toml``:
+  Test Passed!
 
-.. code-block:: text
+  --------------------
 
-    [brick]
-    name = "MyPackage"
-    version = "0.1.0"
-    chplVersion = "1.16.0"
-    authors = ["Sam Partee <Sam@Partee.com>"]
+  --- Results ---
+  Test: myProjectTest Passed
 
-    [dependencies]
-    curl = '1.0.0'
+  --- Summary:  1 tests run ---
+  -----> 1 Passed
+  -----> 0 Failed
 
-The ``chplVersion`` field indicates Chapel releases compatible with this
-package. There are a number of accepted formats:
+Mason will find tests either by searching through the ``test/`` directory, or by
+reading them from the ``Mason.toml`` where they can be specified.
 
-.. code-block:: text
 
-    "1.16.0"         # 1.16.0 or later
-    "1.16"           # 1.16.0 or later
-    "1.16.0..1.19.0" # 1.16 through 1.19, inclusive
+Creating and Running Examples
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-By default, ``chplVersion`` is set to represent the current Chapel release or
-later. For example, if you are using the 1.16 release, chplVersion will be
-``1.16.0``.
+Examples are a great way of showing off what a project can do. Examples are easy
```

Comment:
> Examples are a great way of showing off what a project can do. Examples are easy
to run and configure.

Doesn't add much. Maybe start with "Mason supports examples as a way to demonstrate typical usage of a package. The following example ..."